File: src/main/java/com/mymedia/nagasu/utils/SlugUtils.java
package com.mymedia.nagasu.utils;

public class SlugUtils {

    private SlugUtils() {}

    public static String toSlug(String title) {
        return title.trim()
                .replaceAll("\\s+", "_")
                .replaceAll("[^a-zA-Z0-9가-힣_-]", "");
    }

    public static void validateCollectionId(String collectionId) {
        if (!collectionId.matches("^[a-zA-Z0-9가-힣_-]+$")) {
            throw new IllegalArgumentException(
                    "Invalid collection ID: '" + collectionId + "'. Only alphanumeric, Korean, hyphen(-), and underscore(_) characters are allowed."
            );
        }
    }
}

File: src/main/kotlin/com/mymedia/nagasu/utils/SlugUtils.kt
package com.mymedia.nagasu.utils

fun toSlug(title: String): String {
    return title
        .trim()
        .replace(Regex("\\s+"), "_")
        .replace(Regex("[^a-zA-Z0-9가-힣_-]"), "")
}

fun validateCollectionId(collectionId: String) {
    require(collectionId.matches(Regex("^[a-zA-Z0-9가-힣_-]+$")))
}